packages/shared/src/components/modals/report/ReportSourceModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ReactElement, useRef } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import { Checkbox } from '../../fields/Checkbox';
+import { ModalProps } from '../common/Modal';
+import { useLogContext } from '../../../contexts/LogContext';
+import { ReportModal } from './ReportModal';
+import { ReportReason, sendSourceReport } from '../../../report';
+import { Squad } from '../../../graphql/sources';
+import { useLeaveSquad } from '../../../hooks';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { LogEvent } from '../../../lib/log';
+
+interface SouceReportModalProps extends ModalProps {
+  squad: Squad;
+  onReported?: () => void;
+}
+
+const reportReasons: { value: string; label: string }[] = [
+  { value: ReportReason.Nsfw, label: 'NSFW or inappropriate content' },
+  { value: ReportReason.Spam, label: 'Spam or excessive self-promotion' },
+  { value: ReportReason.Harassment, label: 'Harassment and bullying' },
+  {
+    value: ReportReason.Hateful,
+    label: 'Hate speech and discrimination',
+  },
+  { value: ReportReason.Copyright, label: 'Copyright violation' },
+  { value: ReportReason.Privacy, label: 'Privacy violation' },
+  { value: ReportReason.Miscategorized, label: 'Wrong category' },
+  { value: ReportReason.Illegal, label: 'Illegal activities and scams' },
+  { value: ReportReason.Other, label: 'Other' },
+];
+
+interface SubmitReportProps {
+  reason: ReportReason;
+  comment?: string;
+}
+
+export function ReportPostModal({
+  squad,
+  onReported,
+  onRequestClose,
+  ...props
+}: SouceReportModalProps): ReactElement {
+  const { logEvent } = useLogContext();
+  const { squads } = useAuthContext();
+  const inputRef = useRef<HTMLInputElement>();
+  const onLeaveSquad = useLeaveSquad({ squad });
```

Comment:
Is it specific to squads or also sources?

Reply:
Should probably become applicable to sources too, but should not matter here as we won't show the checkbox if the user is not a member, which is always going to be true.

Reply:
I see, we can start with this then, and modify if we need it for regular sources.